### PR TITLE
[Feature] [WRS-2213] POC support kadanza

### PR DESCRIPTION
**Pull request by abdelhalimkhouas**
This PR is a poc to support multi layout for kadanza's use case

## Related tickets

-   [WRS-2213](https://chilipublishintranet.atlassian.net/browse/WRS-2213)

## Screenshots

[WRS-2213]: https://chilipublishintranet.atlassian.net/browse/WRS-2213?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Demo-POC.html [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+            function toggleMultiLayout() {
+                ToggleSetMultiLayout(true);
+            }
+            function toggleSingleLayout() {
+                ToggleSetMultiLayout(false);
+            }
+        </script>
+    </body>
+</html>
```
Comment (brapoprod) [inline Demo-POC.html:66]: can we not put this in the repo but keep it local?

File: src/App.styles.tsx [hunk 2]
```diff
@@ -4,9 +4,10 @@ import styled from 'styled-components';
 const SANDBOX_NAVBAR_HEIGHT = '3rem';
 const NAVBAR_HEIGHT = '4rem';
 
-export const MainContentContainer = styled.div<{ sandboxMode?: boolean }>`
+export const MainContentContainer = styled.div<{ sandboxMode?: boolean; fullHeight?: boolean }>`
     display: flex;
-    height: ${(props) => `calc(100vh - ${props.sandboxMode ? SANDBOX_NAVBAR_HEIGHT : NAVBAR_HEIGHT})`};
+    height: ${(props) =>
+        props.fullHeight ? `100vh` : `calc(100vh - ${props.sandboxMode ? SANDBOX_NAVBAR_HEIGHT : NAVBAR_HEIGHT})`};
```
Comment (brapoprod) [inline src/App.styles.tsx:10]: 100%; not 100vh

File: src/MainContent.tsx [hunk 3]
```diff
@@ -325,6 +338,9 @@ function MainContent({ projectConfig, updateToken: setAuthToken }: MainContentPr
         loadDocument();
     }, [fetchedDocument]);
 
+    useEffect(() => {
+        zoomToPage();
+    }, [multiLayoutMode, zoomToPage]);
```
Comment (brapoprod) [inline src/MainContent.tsx:343]: only if multilayoutmode is set to false

Comment (abdelhalimkhouas) [reply src/MainContent.tsx:343]: Correct, this is set in the `zoomToPage` function line `124`

File: Demo-POC.html (hunk 1 above)
Comment (abdelhalimkhouas) [reply Demo-POC.html]: done a013f93fd7a091fbf9634e9de7937c75cc695ad3

File: src/App.styles.tsx (hunk 2 above)
Comment (abdelhalimkhouas) [reply src/App.styles.tsx]: done 2e587316daeacd3e2dd7ecbb180b5a89bde74e7f

File: src/MainContent.tsx (hunk 3 above)
Comment (brapoprod) [reply src/MainContent.tsx:343]: From a code cleanness this isn't very clean, zoomToPage indicates a function that should always zoom to page when triggered, please move the check before triggering it

Comment (abdelhalimkhouas) [reply src/MainContent.tsx:338]: done cb480e546f6698f0b3716738fab348e3e8edd31a

**Review by brapoprod — APPROVED** [review]

**Pull request merged**